### PR TITLE
Feature/munjung cnn hit class

### PR DESCRIPTION
**Pull request by wjdanswjddl**
`cnnscore` for PFPs 

- PFP score is the average of the scores of associated clusters -- if there's an outlier among the 3 cluster averages, exclude that cluster from the average
- cluster score is an average of all associated hits -- cluster required to have more than 5 hits
- complementary to existing pfp BDT trackscore
- michel score around PFP end is available, but excluded here for now in interest of CPU time -- plan to replace with mu/pi score in the future
- NN currently stored at `/sbnd/data/users/munjung/nn/CNNID_model.pb` 

dependent on PR#[91](https://github.com/SBNSoftware/sbnobj/pull/91) and PR#[101](https://github.com/SBNSoftware/sbnanaobj/pull/101)

### CODE REVIEW

File: sbncode/TPCReco/CNNHitClassification/CNNID_module.cc [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+        nClusters++;
+
+        cryo = cluster->Plane().Cryostat;
+        view = cluster->Plane().Plane;
+        tpc = cluster->Plane().TPC;
+        fPointIdAlg.setWireDriftData(clockData, detProp, *wireHandle, view, tpc, cryo);
+
+        // hits to apply CNN to (sparsely if nHitsClu > fSparseLengthCut)
+        std::vector<art::Ptr<recob::Hit>> cluHitPtrListApply; 
+        if (nHitsClu > fSparseLengthCut) {
```
Comment (absolution1) [inline sbncode/TPCReco/CNNHitClassification/CNNID_module.cc:217]: It looks like clusters below a certain NHits threshold are ignored.  Is this deliberate?

**Review by absolution1 — APPROVED** [review]
I'm approving the PR.  I have one question related to how some clusters are ignored, but it doesn't need to hold up the PR.

**Comment by absolution1** [comment]
@miquelnebot we'll need to move the NN weights file out of the data directory and into sbnd_data.  Are you the owner of that repo?

Similarly, this PR will need to be checked to ensure the module can still find the weights file.

**Comment by fjnicolas** [comment]
Hi @wjdanswjddl, any updates on this PR??

**Comment by wjdanswjddl** [comment]
> Hi @wjdanswjddl, any updates on this PR??

Hi, I'll test the CI and update this week

**Comment by wjdanswjddl** [comment]
I'll close this and open a new PR as I made major restructuring and updates, as well as tested the code with the most recent code release.

**Pull request closed without merging**